src/main/scala/io/iohk/ethereum/ledger/Ledger.scala

```diff
@@ -389,7 +401,8 @@ sealed trait BlockExecutionError{
 
 object BlockExecutionError {
   case class ValidationBeforeExecError(reason: String) extends BlockExecutionError
-  case class TxsExecutionError(stx: SignedTransaction, reason: String) extends BlockExecutionError
+  case class StateBeforeFailure(worldState: InMemoryWorldStateProxy, acumGas: BigInt, acumReceipts: Seq[Receipt])
+  case class TxsExecutionError(stx: SignedTransaction, stateBeforeError: Option[StateBeforeFailure], reason: String) extends BlockExecutionError
```

Comment:
In which scenario would `stateBeforeError` be `None`? Maybe I missed something but it seems that apart from the use in a test it's used as non empty.